From fa44b37772bff998d3d2d17f3f5f6cd11b20d020 Mon Sep 17 00:00:00 2001
From: Nikos Mavrogiannopoulos <nmav@redhat.com>
Date: Mon, 20 Apr 2015 16:43:51 +0200
Subject: [PATCH 04/19] gnutls_certificate_get_ours: will return the
 certificate even if a callback was used

This corrects a bug where this function would not work, when
gnutls_certificate_set_retrieve_function2() was used.
---
 lib/gnutls_ui.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/gnutls_ui.c b/lib/gnutls_ui.c
index 676a2f3..fb47f99 100644
--- a/lib/gnutls_ui.c
+++ b/lib/gnutls_ui.c
@@ -459,7 +459,7 @@ const gnutls_datum_t *gnutls_certificate_get_ours(gnutls_session_t session)
 
 	cred = (gnutls_certificate_credentials_t)
 	    _gnutls_get_cred(session, GNUTLS_CRD_CERTIFICATE);
-	if (cred == NULL || cred->certs == NULL) {
+	if (cred == NULL) {
 		gnutls_assert();
 		return NULL;
 	}
-- 
2.1.4

